source/MRViewer/MRSpaceMouseHandlerWindows.cpp

```diff
@@ -98,51 +98,55 @@ bool InitializeRawInput()
 
 SpaceMouseHandlerWindows::SpaceMouseHandlerWindows()
 {
+    std::for_each( buttons_.begin(), buttons_.end(), [](auto& v) { v = 0; } );
+    std::for_each( axesOld_.begin(), axesOld_.end(), [](auto& v) { v = 0; } );
+    axesDiff_ = { 0, 0, 0, 0, 0, 0 };
     connect( &getViewerInstance() );
 }
 
+SpaceMouseHandlerWindows::~SpaceMouseHandlerWindows()
+{
+    if ( updateThread_.joinable() )
+        updateThread_.join();
+}
+
 void SpaceMouseHandlerWindows::initialize()
 {
     bool spaceMouseAttached = isSpaceMouseAttached();
     if ( spaceMouseAttached )
-        spdlog::info( "Found attached spacemouse" );
+        spdlog::info( "Found attached SpaceMouse" );
     else
     {
-        spdlog::info( "Not found any attached spacemouse" );
+        spdlog::info( "Not found any attached SpaceMouse" );
         return;
     }
 
     initialized_ = InitializeRawInput();
-    spdlog::info( "InitializeRawInput = {}", initialized_ );
+    spdlog::info( "Initialize SpaceMouse {}", initialized_ ? "success" : "failed" );
 
     updateConnected_();
 }
 
 void SpaceMouseHandlerWindows::handle()
 {
-    if ( !initialized_ || joystickIndex_ == -1 )
+    if ( !initialized_ || joystickIndex_ == -1 || !active_ )
         return;
 
     auto& viewer = getViewerInstance();
     int count;
-    const float* axesNew = glfwGetJoystickAxes( joystickIndex_, &count );
-    if ( count == 6 )
+	std::array<float, 6> axesDiff = axesDiff_;
```

Comment:
We should use diff not as is, but multiply it one the time passed since last application of spacemouse event

Reply:
fixed